src/System.Private.CoreLib/shared/System/IO/BinaryReader.cs

```diff
@@ -649,6 +578,43 @@ public virtual byte[] ReadBytes(int count)
             return result;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
AggressiveInlining should be the other way around: InternalReadSpan should be the method that should be marked with it in the first place since it is smaller and called just from a single place. This method should not be marked with it.

Reply:
fixed